pkg/controller/invite/postInvite.go

```diff
@@ -0,0 +1,182 @@
+	// "github.com/hngprojects/hng_boilerplate_golang_web/external/mocks/send_invites"
+	"github.com/hngprojects/hng_boilerplate_golang_web/external/request"
+	"github.com/hngprojects/hng_boilerplate_golang_web/internal/models"
+	"github.com/hngprojects/hng_boilerplate_golang_web/pkg/repository/storage"
+	"github.com/hngprojects/hng_boilerplate_golang_web/services/invite"
+	"github.com/hngprojects/hng_boilerplate_golang_web/services/organisation"
+	"github.com/hngprojects/hng_boilerplate_golang_web/services/user"
+	"github.com/hngprojects/hng_boilerplate_golang_web/utility"
+)
+
+type Controller struct {
+	Db        *storage.Database
+	Validator *validator.Validate
+	Logger    *utility.Logger
+	ExtReq    request.ExternalRequest
+}
+
+func (base *Controller) PostInvite(c *gin.Context) {
+	var inviteReq models.InvitationRequest
+
+	if err := c.ShouldBindJSON(&inviteReq); err != nil {
+		rd := utility.BuildErrorResponse(http.StatusBadRequest, "error", "Failed to parse request body", err, nil)
+		c.JSON(http.StatusBadRequest, rd)
+		return
+	}
+
+	//check if duplicate emails exist
+	if invite.CheckDuplicateEmails(inviteReq) {
+		rd := utility.BuildErrorResponse(http.StatusConflict, "error", "Duplicate emails found", nil, nil)
+		c.JSON(http.StatusConflict, rd)
+		return
+	}
+
+	// check emails limit
+	if invite.CheckEmailsLimit(inviteReq) {
+		rd := utility.BuildErrorResponse(http.StatusRequestEntityTooLarge, "error", "Payload too large; email limit exceeded", nil, nil)
+		c.JSON(http.StatusBadRequest, rd)
+		return
+	}
+
+	err := base.Validator.Struct(&inviteReq)
+	if err != nil {
+		rd := utility.BuildErrorResponse(http.StatusBadRequest, "error", "Validation failed", utility.ValidationResponse(err, base.Validator), nil)
+		c.JSON(http.StatusBadRequest, rd)
+		return
+	}
+
+	// Validate org_id
+	orgId, err := uuid.Parse(inviteReq.OrgID)
+	if err != nil {
+		rd := utility.BuildErrorResponse(http.StatusUnprocessableEntity, "error", "Invalid org_id format", err, nil)
+		c.JSON(http.StatusUnprocessableEntity, rd)
+		return
+	}
+
+	// Check if org_id exists and return organization
+	org, err := organisation.CheckOrgExists(orgId.String(), base.Db.Postgresql)
+	if err != nil {
+		rd := utility.BuildErrorResponse(http.StatusNotFound, "error", "organisation not found", err, nil)
+		c.JSON(http.StatusNotFound, rd)
+		return
+	}
+
+	///check if user from the claims is a member of the organisation
+	claims, exists := c.Get("userClaims")
+	if !exists {
+		rd := utility.BuildErrorResponse(http.StatusBadRequest, "error", "unable to get user claims", nil, nil)
+		c.JSON(http.StatusBadRequest, rd)
+		return
+	}
+
+	userClaims := claims.(jwt.MapClaims)
+	userId := userClaims["user_id"].(string)
+
+	// Check if user is a member of the organization
+	isMember, err := organisation.CheckUserIsMemberOfOrg(userId, org.ID, base.Db.Postgresql)
+	if err != nil {
+		rd := utility.BuildErrorResponse(http.StatusNotFound, "error", "User not a member of the organization", err, nil)
```

Comment:
only request validation should be on your controller, please look at what others did

Reply:
.